src/libaktualizr/package_manager/dockerappmanager.cc

```diff
@@ -111,6 +122,24 @@ data::InstallationResult DockerAppManager::install(const Uptane::Target &target)
   return res;
 }
 
+// Handle the case like:
+//  1) sota.toml is configured with 2 docker apps: "app1, app2"
+//  2) update is applied, so we are now running both app1 and app2
+//  3) sota.toml is updated with 1 docker app: "app1"
+// At this point we should stop app2 and remove it.
+void DockerAppManager::handleRemovedApps() const {
+  for (auto &entry : boost::make_iterator_range(boost::filesystem::directory_iterator(config.docker_apps_root), {})) {
+    if (boost::filesystem::is_directory(entry)) {
+      std::string name = entry.path().filename().native();
+      if (std::find(config.docker_apps.begin(), config.docker_apps.end(), name) == config.docker_apps.end()) {
+        LOG_WARNING << "Docker App(" << name
```

Comment:
Should this just be info instead of warning?

Reply:
I'm ok with the warning here.  It's something to log when a user removes an app this way (IMHO).

Reply:
I debated the same thing. I'm fine with LOG_INFO. I'll change it.

Reply:
good point mike. i'll leave it as a warning. maybe when i upstrea this change, the ATS guys will have opinions.